list_sorting.py

```diff
@@ -0,0 +1,34 @@
+#function to sort elements according to evens, odds and characters
+def list_sort(a):
+    if isinstance(a, list):
+        #list categories
+        word_cat = ['evens','odds','chars']
+        #sorted lis
+        sorted_list = {}
+        char_list = []
+        evens_list = [] 
+        odds_list = []
+        for x in a:
+            if isinstance(x, str):
+                char_list.append(x)
+                sorted_list[word_cat[2]] = char_list
+                char_list.sort()
+            elif x % 2 == 1:
+                evens_list.append(x)
+                sorted_list[word_cat[1]] = evens_list
```

Comment:
sorting and updating the list can be moved out of the loop and done once the loop has finished